echo_server.py

```diff
@@ -29,5 +33,39 @@ def start_server():
     finally:
         raise
 
+
+def response_ok(uri):
+    okay_response = "HTTP/1.1 200 OK\r\n"
+    headers = "Content-Type: text/plain\r\n"
+    body = "you requested: {}".format(uri)
+    return "{}{}{}\r\n".format(okay_response, headers, body)
+
+
+def response_error(error_num, error_msg):
+    error_response = "HTTP/1.1 {} ERROR\r\n".format(error_num)
+    headers = "Content-Type: text/plain\r\n"
+    body = "ERROR {}, {}\r\n".format(error_num, error_msg)
+    return "{}{}{}".format(error_response, headers, body)
+
+
+def parse_request(request):
+    request_list = request.split()
+    print len(request_list)
+    if len(request_list) != 5:
+        return None
+    error_check = check_for_errors(request_list)
+    if error_check == "Good to go!":
+        return response_ok(request_list[1])
+    return error_check
```

Comment:
the specs say that this function should return a URI.  I see three possible return values here:  `None`, an error http response and an ok http response.  None of those are a URI.